packages/git-encrypted/src/encryptedMeta.ts

```diff
@@ -0,0 +1,96 @@
+import zod from 'zod';
+import { superpathjoin as join } from 'superpathjoin';
+
+import { FS, GitBaseParamsEncrypted } from './types';
+import { generateKey } from './crypto';
+import { encryptedRepoCommit } from './git';
+
+const META_FILENAME = 'encrypted.json';
+
+const EncryptedMetaSchema = zod.object({
+  version: zod.literal(1),
+  derivationParams: zod.object({
+    salt: zod.string().nonempty(),
+    cpuCost: zod
+      .number()
+      .int()
+      .positive(),
+    blockSize: zod
+      .number()
+      .int()
+      .positive(),
+    parallelizationCost: zod
+      .number()
+      .int()
+      .positive(),
+  }),
+});
+type EncryptedMeta = zod.infer<typeof EncryptedMetaSchema>;
+
+const createEncryptedMeta = (): EncryptedMeta => {
+  return {
+    version: 1,
+    derivationParams: {
+      salt: generateKey(1024),
+      cpuCost: 1024,
+      blockSize: 8,
+      parallelizationCost: 1,
+    },
+  };
+};
+
+const getMetaPath = ({ encryptedDir }: { encryptedDir: string }) => {
+  return join(encryptedDir, META_FILENAME);
+};
+
+const readMetaFromDisk = async ({
+  fs,
+  encryptedDir,
+}: {
+  fs: FS;
+  encryptedDir: string;
+}): Promise<EncryptedMeta> => {
+  console.log('#SbXtKQ', fs, encryptedDir);
```

Comment:
nit: This was a mock to stop TS complaining about the params being unused, we can remove it.